src/almanac.js

```diff
@@ -122,44 +127,19 @@ export default class Almanac {
         factValuePromise = this._setFactValue(fact, params, fact.calculate(params, this))
       }
     }
-    if (path) { // selectn supports arrays and strings as a 'path'
-      // strings starting with '$' denotes json path. otherwise fall back to deprecated 'selectn' syntax
-      if (typeof path === 'string' && path.startsWith('$')) {
-        debug(`condition::evaluate extracting object property ${path}`)
-        return factValuePromise
-          .then(factValue => {
-            if (isObjectLike(factValue)) {
-              const pathValue = JSONPath({ path, json: factValue, wrap: false })
-              debug(`condition::evaluate extracting object property ${path}, received: ${JSON.stringify(pathValue)}`)
-              return pathValue
-            } else {
-              debug(`condition::evaluate could not compute object path(${path}) of non-object: ${factValue} <${typeof factValue}>; continuing with ${factValue}`)
-              return factValue
-            }
-          })
-      } else {
-        let selectn
-        try {
-          selectn = require('selectn')
-        } catch (err) {
-          console.error('Oops! Looks like you\'re trying to use the deprecated syntax for the ".path" property.')
-          console.error('Please convert your "path" properties to JsonPath syntax (ensure your path starts with "$")')
-          console.error('Alternatively, if you wish to continue using old syntax (provided by selectn), you may "npm install selectn" as a direct dependency.')
-          console.error('See https://github.com/CacheControl/json-rules-engine/blob/master/CHANGELOG.md#500--2019-10-27 for more information.')
-          throw new Error('json-rules-engine: Unmet peer dependency "selectn" required for use of deprecated ".path" syntax. please "npm install selectn" or convert to json-path syntax')
-        }
-        return factValuePromise
-          .then(factValue => {
-            if (isObjectLike(factValue)) {
-              const pathValue = selectn(path)(factValue)
-              debug(`condition::evaluate extracting object property ${path}, received: ${pathValue}`)
-              return pathValue
-            } else {
-              debug(`condition::evaluate could not compute object path(${path}) of non-object: ${factValue} <${typeof factValue}>; continuing with ${factValue}`)
-              return factValue
-            }
-          })
-      }
+    if (path) {
+      debug(`condition::evaluate extracting object property ${path}`)
+      return factValuePromise
+        .then(factValue => {
+          if (isObjectLike(factValue)) {
+            const pathValue = this.pathResolver(factValue, path)
```

Comment:
Do you think it makes sense pass to `params` as well ? Just in case, some custom `pathResolver ` need details from params.
```suggestion
            const pathValue = this.pathResolver(factValue, path, params)
```

Reply:
oh! That's an interesting idea and I see where you're coming from - this could allow a user to configure their path resolve library differently on a condition by condition basis. 🤔 

My immediate thinking is that we don't want to mix the concept of `fact parameters` with `path resolver parameters` - as things stand right now there would be unintended consequences to things like fact caching, which uses `params` for fact idempotency. That said, we could easily avoid that issue with a `pathParams` property.

For now, I think I'd like to see if this request actually materializes in the wild before I add it, and then it can be introduced as a minor version. But cool idea.